### PR TITLE
Rename hidden py4design functions in geometry_generator

### PR DESCRIPTION
**Pull request by martin-mosteiro**
There are a number of functions in `geometry_generator` that are hidden as comments but are useful for debugging. These use outdated py4design function names, though, so I have updated all the names in the hidden functions.

### CODE REVIEW

**Comment by daren-thomas** [comment]
@gabriel-happle this is a good example of why i don't like commenting out code to keep developer-only features too much and prefer the `debug` flag or similar. The original refactoring effort would otherwise have picked this up.

@martin-mosteiro how do we test this? is there a way to make the commented stuff turn on with the `debug` flag? or would you suggest adding another flag maybe? This is also a good time to add some documentation to this module. I was lacking info on _why_ we need this? and the main way it's done. i mean, i'm _guessing_ it's just done by extruding the footprint based on level height * level count?

**Comment by martin-mosteiro** [comment]
@daren-thomas I think the debug flag(s) is a good idea. I'll add them to this PR in the next few days.

**Comment by JIMENOFONSECA** [comment]
@martin-mosteiro try the local tests before pushing ;-)

image

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA I didn't do the tests because my original PR only changed commented lines. Nevertheless I'm not sure why the tests should fail if none of the actual code gets changed.

File: cea/resources/radiation_daysim/geometry_generator.py [hunk 1]
```diff
@@ -401,8 +401,8 @@ def geometry_main(locator, simplification_params):
     geometry_buildings.extend(windows_s)
     geometry_buildings.extend(roof_s)
 
-    # DO this to visualize progress while debugging!:
-    #normals_terrain = calculate.visualise_face_normal_as_edges(geometry_terrain,5)
+    # # DO this to visualize progress while debugging!:
+    # normals_terrain = calculate.face_normal_as_edges(geometry_terrain,5)
     utility.visualise([geometry_terrain, geometry_buildings], ["BLUE","WHITE"]) #install Wxpython
```
[the remark this answers is not in the document]
Comment (martin-mosteiro) [reply cea/resources/radiation_daysim/geometry_generator.py]: That line was not commented in the original code. I only changed the coments.

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA This is strange. I reverted all changes that were not directly related to this issue in order to check why the tests were not passing. Ran the tests, got the following:
```
[...]
.  run_sensitivity
########################################
TaskError - taskid:run_sensitivity
PythonAction Error
Traceback (most recent call last):
  File "C:\Users\User\Miniconda2\envs\cea\lib\site-packages\doit\action.py", line 383, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "C:/Users/User/Documents/GitHub/CityEnergyAnalyst/cea/tests/dodo.py", line 303, in run_sensitivity
    cea.analysis.sensitivity.sensitivity_demand_simulate.main(config)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\analysis\sensitivity\sensitivity_demand_simulate.py", line 238, in main
    output_parameters=config.sensitivity_demand.output_parameters)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\analysis\sensitivity\sensitivity_demand_simulate.py", line 186, in simulate_demand_batch
    totals, time_series = simulate_demand_sample(locator, config, output_parameters)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\analysis\sensitivity\sensitivity_demand_simulate.py", line 132, in simulate_demand_sample
    totals, time_series = demand_main.demand_calculation(locator, config)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\demand\demand_main.py", line 111, in demand_calculation
    format_output, config, region,  write_detailed_output, debug)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\demand\demand_main.py", line 155, in calc_demand_singleprocessing
    format_output, config, region,  write_detailed_output, debug)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\demand\thermal_loads.py", line 118, in calc_thermal_loads
    tsd = sensible_loads.calc_temperatures_emission_systems(bpr, tsd) # calculate temperatures
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\demand\sensible_loads.py", line 274, in calc_temperatures_emission_systems
    Ta_sup_0, Ta_re_0)
  File "C:\Users\User\Miniconda2\envs\cea\lib\site-packages\numpy\lib\function_base.py", line 1972, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "C:\Users\User\Miniconda2\envs\cea\lib\site-packages\numpy\lib\function_base.py", line 2048, in _vectorize_call
    outputs = ufunc(*inputs)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\technologies\heating_coils.py", line 62, in calc_heating_coil
    dTm_0 = substation.calc_dTm_HEX(thi_0, tho_0, tci_0, tco_0)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\technologies\substation.py", line 768, in calc_dTm_HEX
    raise Exception(thi, tco, tho, tci, "Check the emission_system database, there might be a problem with the selection of nominal temperatures")
Exception: (313.0, 309.0, 293.0, 293.0, 'Check the emission_system database, there might be a problem with the selection of nominal temperatures')
[...]
Process finished with exit code 2
```

Didn't change the code and ran the tests. Got the following:
```
.  run_sensitivity
.  run_unit_tests
...........................
----------------------------------------------------------------------
Ran 27 tests in 76.540s

OK

Process finished with exit code 0
```

@daren-thomas Is it possible that the sensitivity analysis test sometimes passes and sometimes doesn't!?

**Comment by martin-mosteiro** [comment]
I added the heatmap commit again and ran the tests. Same deal: first time they failed, second time they passed.

**Comment by martin-mosteiro** [comment]
I added the benchmark part and ran the dodo again. All tests pass.

**Comment by martin-mosteiro** [comment]
Fixed an issue in the benchmark plots legend. All tests pass locally.

**Comment by martin-mosteiro** [comment]
Fixed an issue in the benchmark plots legend. All tests pass locally.
This PR now does the following:
1 - updated the references to py4design so that the visualization functionality in the `geometry_generator` are restored
2 - added the "debug" flag to the geometry generator so the user can choose whether to visualize building geometries for debugging purposes or not
3 - added `general:debug` as a parameter for `radiation-daysim` in `scripts.yml`
4 - restored functionality to the `heatmaps` script in legacy - this script was moved to legacy since we were moving away from ArcGIS, however for those of us who do have ArcGIS it's nice to have it
5 - fixed the benchmark graphs and database which weren't working anymore

None of these affect core functionalities to CEA, so they shouldn't affect the tests at all.

@JIMENOFONSECA All tests passed locally. From my testing, however, it seems that the tests sometimes pass and sometimes don't. I think there is an issue in the tests, and namely I think the sensitivity analysis test might not be producing consistent results.

**Comment by martin-mosteiro** [comment]
The error message itself, however, points to the heat exchanger model I worked on last week or so. The nominal outlet temperature on the hot side seems to be the same as the nominal inlet temperature in the cold side. This has as of yet never come up in my own simulations, but it might be worth finding a solution for this for other cases.

**Comment by JIMENOFONSECA** [comment]
Hi martin,

I got tge same error message. Might this be related with changes in tge testcases of cea?

_______________________

Dr. Jimeno A. Fonseca
Senior researcher and project coordinator
Future cities Laboratory
Singapore ETH-Centre

Chair of Architecture and Building Systems
ETH Zurich

Product owner
City Energy Anlayst (CEA)<http://www.cityenergyanalyst.com>

Prof.h.c in modeling and simulation
Universidad Central de Colombia
On 8 Mar 2019, 01:39 +0800, Martín Mosteiro-Romero <notifications@github.com>, wrote:

The error message itself, however, points to the heat exchanger model I worked on last week or so. The nominal outlet temperature on the hot side seems to be the same as the nominal inlet temperature in the cold side. This has as of yet never come up in my own simulations, but it might be worth finding a solution for this for other cases.

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA You're right, it's due to the new reference-case-open. In summary, the sensitivity analysis tests _sometimes_ fail because _sometimes_ a building with heating system `T3` might be assigned a heating set point of 20°C, which _might_ lead to a situation where the nominal return temperature in the heating coil is equal to the nominal return temperature from the room (20°C) and hence there is no heat exchange and the heat exchanger model fails.

It is worth mentioning that an air-based heating system with operating temperatures of 40°/20° is indeed very low - probably uncommon since there will generally never be any heat exchange at 20°C. So I see two options:
1- Assume a slightly hotter emission system for centralized AC heating
2- Add a flag to deal with nominal temperatures that the current model cannot handle
3- ???

**Comment by JIMENOFONSECA** [comment]
Hi Martin,

I will go for 1 and 2.

That return temperature is definately a mistake.

Perhaps a default return of 25 degrees could also work for all cases?

On 9 Mar 2019, 01:07 +0800, Martín Mosteiro-Romero <notifications@github.com>, wrote:

@JIMENOFONSECA<https://github.com/JIMENOFONSECA> You're right, it's due to the new reference-case-open. In summary, the sensitivity analysis tests sometimes fail because sometimes a building with heating system T3 might be assigned a heating set point of 20°C, which might lead to a situation where the nominal return temperature in the heating coil is equal to the nominal return temperature from the room (20°C) and hence there is no heat exchange and the heat exchanger model fails.

It is worth mentioning that an air-based heating system with operating temperatures of 40°/20° is indeed very low - probably uncommon since there will generally never be any heat exchange at 20°C. So I see two options:
1- Assume a slightly hotter emission system for centralized AC heating
2- Add a flag to deal with nominal temperatures that the current model cannot handle
3- ???

**Comment by daren-thomas** [comment]
@JIMENOFONSECA @martin-mosteiro what is the current status?

**Comment by martin-mosteiro** [comment]
@daren-thomas The PR does exactly what it's supposed to do, but the tests revealed an unrelated issue with the emissions database. A solution has been found and needs to be implemented.

**Comment by martin-mosteiro** [comment]
All tests pass locally. Adding @shanshanhsieh as a reviewer since there is a minor change to the heat exchanger model.

**Comment by martin-mosteiro** [comment]
@daren-thomas All tests pass locally, but the jenkins disapproves. Maybe we can look at it tomorrow?

**Pull request merged**